ftw/iframeblock/browser/templates/iframeblock.pt

```diff
@@ -11,14 +11,32 @@
             width="100%"
             class="iframeblock loading" onload="onIframeLoaded(this)"></iframe>
     <script tal:condition="context/auto_size">
-      var heightCalculationMethod = $("iframe.iframeblock").data("heightCalculationMethod");
-      iFrameResize({
-        inPageLinks: true,
-        heightCalculationMethod: heightCalculationMethod,
-        resizedCallback: function () {
-          scroll(0, 0);
-        }
-      })
+      if (typeof require === 'function') {
+        require([
+          'jquery',
+          'iframeResizer'
+        ], function($, iFrameResize){
+          $(document).ready(function() {
```

Comment:
`$(document).ready` should be just as applicable without requirejs.

I wonder if something like the following pattern would stop the code duplication

    (function (factory) {
        // environment detection
        if (typeof define === 'function' && define.amd) {
            // AMD. Register as an anonymous module.
            require(['jquery', 'iframeResizer'], factory);
        } else {
            // Browser globals
            factory(window.jQuery, window.iFrameResize);
        }
    }(function ($, iFrameResize) {
        // Use $ and iFrameResize here
    
    }));

It would also be good to stop the JS in page templates at some point

Reply:
I did not test the iframeblock without your suggestion, but with it the following errors occur with the embedded url: https://baupub.bern.ch/stadtverwaltung/prd/bauinspektorat/pub_list_contents

![image](https://user-images.githubusercontent.com/29920936/70734947-7ef53180-1d0d-11ea-8895-76f983775091.png)

Reply:
```
		function postMessageToIFrame(){
			var target = settings[id].targetOrigin;
			log(id,'[' + calleeMsg + '] Sending msg to iframe['+id+'] ('+msg+') targetOrigin: '+target);
      if(calleeMsg === 'init') {
        iframe.contentWindow.postMessage( msgId + msg, '*' );
      } else {
			  iframe.contentWindow.postMessage( msgId + msg, target );
      }
		}
```

This approach for https://github.com/4teamwork/ftw.iframeblock/blob/0f9ef61dc8c02ccbf220d2006e66980f438d8260/ftw/iframeblock/browser/resources/iframeResizer.js#L623-L627 would already make the above error away. The solution is not yet clean though since it raises the warning below now:

![image](https://user-images.githubusercontent.com/29920936/70744214-8e7e7580-1d21-11ea-9f06-8ae1e8de68b3.png)

I'm not yet sure how to solve this one and if it wouldn't require the header `SameSite=None` from the embedded site...

Reply:
In 00e4a1e I found an approach for fixing the error. The iframeReload method gets now called later which makes the error disappear. Also see: https://github.com/davidjbradshaw/iframe-resizer/issues/443#issuecomment-331721886.

Reply:
The warning is pretty important in the near future so we probably should also fix it -> https://blog.chromium.org/2019/10/developers-get-ready-for-new.html
I'm looking into that right now.

Reply:
@bierik, if you are interested this is the stuff I was asking you this morning...

Reply:
@bierik, if you are interested this is the stuff I was asking you this morning...